chasm/field.go

```diff
@@ -26,29 +26,48 @@ package chasm
 
 import (
 	"reflect"
+	"strings"
 
 	"google.golang.org/protobuf/proto"
 )
 
+const (
+	// Used by reflection.
+	chasmFieldTypePrefix      = "chasm.Field["
+	chasmCollectionTypePrefix = "chasm.Collection["
+	internalFieldName         = "Internal"
+
+	fieldNameTag = "name"
+)
+
 // This struct needs to be create via reflection
 // but reflection can't set prviate fields...
 type Field[T any] struct {
 	Internal fieldInternal
 }
 
 type fieldInternal struct {
-	fieldType int // data, component, componentPointer
-	component reflect.Value
+	fieldType fieldType
+	value     any // Component | Data | Collection
 
-	// backingNode *nodeInfo
+	// Pointer to corresponding tree node. Can be nil for the jsut assigned fields.
+	node *Node
 }
 
-func (d *Field[T]) Get(Context) (T, error) {
+func (fi fieldInternal) IsEmpty() bool {
+	return fi.value == nil
+}
+
+func (f Field[T]) Get(Context) (T, error) {
 	// remember to handle d == nil case
 
 	panic("not implemented")
 }
 
+func NewEmptyField[T any]() Field[T] {
+	return Field[T]{}
+}
```

Comment:
nit: not sure how useful this is. Component author can just do Field[T]{} themselves to clear/delete a field.

Reply:
I think it looks better with constructor. It's a matter of personal preferences though.